src/bindgen/mangle.rs

```diff
@@ -4,31 +4,35 @@
 
 use crate::bindgen::ir::{Path, Type};
 
-pub fn mangle_path(path: &Path, generic_values: &[Type]) -> Path {
-    internal_mangle_path(path, generic_values, true)
+pub fn mangle_path(path: &Path, generic_values: &[Type], mangle_separator: &Option<String>) -> Path {
+    internal_mangle_path(path, generic_values, true, mangle_separator)
 }
 
-pub fn mangle_name(name: &str, generic_values: &[Type]) -> String {
-    internal_mangle_name(name, generic_values, true)
+pub fn mangle_name(name: &str, generic_values: &[Type], mangle_separator: &Option<String>) -> String {
+    internal_mangle_name(name, generic_values, true, mangle_separator)
 }
 
-fn internal_mangle_path(path: &Path, generic_values: &[Type], last_in_parent: bool) -> Path {
+fn internal_mangle_path(path: &Path, generic_values: &[Type], last_in_parent: bool, mangle_separator: &Option<String>) -> Path {
     let name = path.name();
-    let mangled_name = internal_mangle_name(name, generic_values, last_in_parent);
+    let mangled_name = internal_mangle_name(name, generic_values, last_in_parent, mangle_separator);
     Path::new(mangled_name)
 }
 
-fn internal_mangle_name(name: &str, generic_values: &[Type], last_in_parent: bool) -> String {
+fn internal_mangle_name(name: &str, generic_values: &[Type], last_in_parent: bool, mangle_separator: &Option<String>) -> String {
     if generic_values.is_empty() {
         return name.to_owned();
     }
 
     let mut mangled = name.to_owned();
+    let separator = match mangle_separator {
+        Some(s) => s.to_owned(),
```

Comment:
If this was an `Option<&str>`, then you don't need to heap-allocate here.

You don't need here anyway though, this could be:

```
match mangle_separator {
    Some(s) => &**s,
    None => "_",
}
```